redbeat/schedulers.py

```diff
@@ -365,14 +365,15 @@ class RedBeatScheduler(Scheduler):
 
     def __init__(self, app, lock_key=None, lock_timeout=None, **kwargs):
         ensure_conf(app)  # set app.redbeat_conf
+        super(RedBeatScheduler, self).__init__(app, **kwargs)
+
         self.lock_key = lock_key or app.redbeat_conf.lock_key
         self.lock_timeout = (
             lock_timeout
             or app.redbeat_conf.lock_timeout
             or self.max_interval * 5
             or self.lock_timeout
         )
-        super(RedBeatScheduler, self).__init__(app, **kwargs)
```

Comment:
It seems weird that this super() call was towards the end of the constructor rather than the start.  I want to mull this over a little to see if I can recall why I did that.  Unless I can recall some reason for the delayed super() call I'll merge this shortly.

Reply:
looking back at, https://github.com/sibson/redbeat/commit/91d7f84e3dffc3e82731ea42c2b0fc1a95da2ab2#diff-3306d7d435bbd01ef84d83cae122c8cfR142-R148, my guess is that the Scheduler.__init__() in older Celery versions didn't gracefully handle unexpected values in kwargs.   Further I'm going to assume that subsequent changes to configuration handling mean that we no longer need to pop() values out of kwargs before calling __init__().

Reply:
10-4!